### PR TITLE
Disable prefer partial aggregation for `SHOW STATS` queries

### PR DESCRIPTION
**Pull request by wendigo**
Fixes https://github.com/trinodb/trino/issues/6323

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/Session.java [hunk 1]
```diff
@@ -529,7 +529,13 @@ public static SessionBuilder builder(SessionPropertyManager sessionPropertyManag
     @VisibleForTesting
     public static SessionBuilder builder(Session session)
     {
-        return new SessionBuilder(session);
+        return new SessionBuilder(session, false);
+    }
+
+    @VisibleForTesting
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/Session.java:535]: This annotation is not needed, it is used by product code.

File: core/trino-main/src/main/java/io/trino/Session.java [hunk 2]
```diff
@@ -529,7 +529,13 @@ public static SessionBuilder builder(SessionPropertyManager sessionPropertyManag
...
+    public static SessionBuilder builderWithCopyTransactionId(Session session)
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/Session.java:536]: `builderFromTransaction`?

File: core/trino-main/src/main/java/io/trino/Session.java [hunk 3]
```diff
@@ -568,10 +574,18 @@ private SessionBuilder(SessionPropertyManager sessionPropertyManager)
...
 
-        private SessionBuilder(Session session)
+        private SessionBuilder(Session session, boolean copyTransactionId)
         {
             requireNonNull(session, "session is null");
-            checkArgument(session.getTransactionId().isEmpty(), "Session builder cannot be created from a session in a transaction");
+
+            if (copyTransactionId) {
+                session.getTransactionId()
+                        .ifPresent(transactionId -> this.transactionId = transactionId);
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/Session.java:583]: let's fail if there is no transaction

File: core/trino-main/src/main/java/io/trino/sql/rewrite/ShowStatsRewrite.java [hunk 4]
```diff
@@ -185,6 +186,13 @@ protected Node visitNode(Node node, Void context)
             return node;
         }
 
+        private static Session disablePartialAggregation(Session session)
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/sql/rewrite/ShowStatsRewrite.java:189]: Please add a comment why

**Review by kokosing — COMMENTED** [review]
Some minor comments

**Comment by wendigo** [comment]
PTAL @kokosing

**Comment by electrum** [comment]
Why was this closed? Based on the tests, it seems to fix the issue.

**Comment by wendigo** [comment]
@electrum we've decided to pursue other approach

**Pull request closed without merging**